### PR TITLE
Revert the initialization of evict-slow-trend scheduler in `init_schedulers`.

### PR DESCRIPTION
**Pull request by LykxSassinator**
### What problem does this PR solve?

Issue Number: ref #6981 

### What is changed and how does it work?

```commit-message
In this pr:
+ Revert and re-factor the initialization of `evict-slow-trend` scheduler when starting with `raft-kv2`.
+ Fix typo errors.
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

File: pkg/schedule/coordinator.go [hunk 1]
```diff
@@ -346,12 +381,14 @@ func (c *Coordinator) Run() {
...
 	c.initSchedulers()
 
-	c.wg.Add(3)
+	c.wg.Add(4)
 	// Starts to patrol regions.
 	go c.PatrolRegions()
 	// Checks suspect key ranges
 	go c.checkSuspectRanges()
 	go c.drivePushOperator()
+	// // Checks whether to create evict-slow-trend scheduler.
```
Comment (HuSharp) [inline pkg/schedule/coordinator.go:390]:
```suggestion
	// Checks whether to create evict-slow-trend scheduler.
```

File: pkg/schedule/coordinator.go [hunk 2]
```diff
@@ -313,6 +312,42 @@ func (c *Coordinator) drivePushOperator() {
...
+			return
+		case <-ticker.C:
+			{
+				// If enabled, exit.
+				if exists, _ := c.schedulers.IsSchedulerExisted(schedulers.EvictSlowTrendName); exists {
+					return
+				}
+				// If the cluster was set up with `raft-kv2` engine, this cluster should
+				// enable `evict-slow-trend` scheduler as default.
+				if c.GetCluster().GetStoreConfig().IsRaftKV2() {
```
Comment (nolouch) [inline pkg/schedule/coordinator.go:335]: this is unbelievable

Comment (LykxSassinator) [reply pkg/schedule/coordinator.go:335]: Nope, it will not influence the current strategy when initializing.

**Comment by nolouch** [comment]
/merge

BTW, We need unify the bootstrap method. we can default enable this scheduler, and judge `IsRaftKV` inner the scheduler. current implement will let some issues like if user manually delete the slowTrendScheduler, after restart, the scheduler will be recreated.

**Pull request merged**